lib/ui/survey_question/answer/answer_emoji_rating.dart

```diff
@@ -0,0 +1,88 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:survey_flutter_ic/extension/toast_extension.dart';
+import 'package:survey_flutter_ic/model/survey_answer_model.dart';
+import 'package:survey_flutter_ic/model/survey_question_model.dart';
+import 'package:survey_flutter_ic/theme/dimens.dart';
+
+const defaultSelectedEmojiIndex = 2;
+const maxAnswerChoices = 5;
+const List<String> faceModes = ['😡', '😕', '😐', '🙂', '😄'];
+
+final selectedEmojiIndexProvider =
+    StateProvider.autoDispose<int>((_) => defaultSelectedEmojiIndex);
+
+class AnswerEmojiRating extends ConsumerWidget {
+  final DisplayType displayType;
+  final List<SurveyAnswerModel> answers;
+
+  const AnswerEmojiRating({
+    super.key,
+    required this.displayType,
+    required this.answers,
+  });
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final selectedAnswerIndex = ref.watch(selectedEmojiIndexProvider);
+
+    return ListView.separated(
+      shrinkWrap: true,
+      scrollDirection: Axis.horizontal,
+      itemCount: maxAnswerChoices,
+      itemBuilder: (context, index) {
+        return GestureDetector(
+          onTap: () {
+            ref.read(selectedEmojiIndexProvider.notifier).state = index;
+            // TODO: Trigger VM on Integration of submit task and reset index to default
+            showToastMessage(answers[index].text);
+          },
+          child: Center(
+            child: Text(
+              getEmoji(displayType, index),
+              style: TextStyle(
+                color: getColor(displayType, index, selectedAnswerIndex),
+                fontSize: fontSize34,
+              ),
+            ),
+          ),
+        );
+      },
+      separatorBuilder: (_, __) {
+        return const SizedBox(width: space20);
+      },
+    );
+  }
+
+  String getEmoji(
```

Comment:
Should this be `private`?

```suggestion
  String _getEmoji(
```